Build failed when com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension is enabled

When Spring Swagger is turned on in pom.xml, running mvn clean package will failed.  Here is the snippet of the pom.xml

               "${basedir}/src/main/resources/template/hello.html" -->
              <swaggerApiReader>com.github.kongchen.swagger.docgen.reader.SpringMvcApiReader</swaggerApiReader>
              <swaggerExtensions>
                <swaggerExtension>com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension</swaggerExtension>
              </swaggerExtensions>
              <modelConverters>
                <modelConverter>io.swagger.validator.BeanValidator</modelConverter>
              </modelConverters>
              <swaggerDirectory>${project.basedir}/src/main/webapp/api-spec/v1</swaggerDirectory>
              <swaggerFileName>swagger</swaggerFileName>
              <outputFormats>
                json,yaml
              </outputFormats>
              <attachSwaggerArtifact>true</attachSwaggerArtifact>

[ERROR] Failed to execute goal com.github.kongchen:swagger-maven-plugin:3.1.8:generate (default) on project imsrestapi-web: Cannot load Swagger extension: com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension: com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension.<init>() -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal com.github.kongchen:swagger-maven-plugin:3.1.8:generate (default) on project imsrestapi-web: Cannot load Swagger extension: com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:212)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.MojoFailureException: Cannot load Swagger extension: com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension
        at com.github.kongchen.swagger.docgen.mavenplugin.ApiDocumentMojo.execute(ApiDocumentMojo.java:150)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
Caused by: com.github.kongchen.swagger.docgen.GenerateException: Cannot load Swagger extension: com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension
        at com.github.kongchen.swagger.docgen.AbstractDocumentSource.resolveSwaggerExtensions(AbstractDocumentSource.java:438)
        at com.github.kongchen.swagger.docgen.AbstractDocumentSource.loadSwaggerExtensions(AbstractDocumentSource.java:162)
        at com.github.kongchen.swagger.docgen.AbstractDocumentSource.loadDocuments(AbstractDocumentSource.java:106)
        at com.github.kongchen.swagger.docgen.mavenplugin.ApiDocumentMojo.execute(ApiDocumentMojo.java:122)
        ... 22 more
Caused by: java.lang.InstantiationException: com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension
        at java.lang.Class.newInstance(Class.java:427)
        at com.github.kongchen.swagger.docgen.AbstractDocumentSource.resolveSwaggerExtensions(AbstractDocumentSource.java:436)
        ... 25 more
Caused by: java.lang.NoSuchMethodException: com.github.kongchen.swagger.docgen.spring.SpringSwaggerExtension.<init>()
        at java.lang.Class.getConstructor0(Class.java:3082)
        at java.lang.Class.newInstance(Class.java:412)
        ... 26 more

--- a/src/main/java/com/github/kongchen/swagger/docgen/AbstractDocumentSource.java
+++ b/src/main/java/com/github/kongchen/swagger/docgen/AbstractDocumentSource.java
@@ -431,10 +431,17 @@ public abstract class AbstractDocumentSource {
         List<SwaggerExtension> resolved = new ArrayList<SwaggerExtension>();
         if (clazzes != null) {
             for (String clazz : clazzes) {
-                SwaggerExtension extension;
+                SwaggerExtension extension = null;
+                //Try to get a parameterized constructor for extensions that are log-enabled.
                 try {
-                    extension = (SwaggerExtension) Class.forName(clazz).newInstance();
-                } catch (Exception e) {
+                    try {
+                        Constructor<?> constructor = Class.forName(clazz).getConstructor(Log.class);
+                        extension = (SwaggerExtension) constructor.newInstance(LOG);
+                    } catch (NoSuchMethodException nsme) {
+                        extension = (SwaggerExtension) Class.forName(clazz).newInstance();
+                    }
+                }
+                catch (Exception e) {
                     throw new GenerateException("Cannot load Swagger extension: " + clazz, e);
                 }
                 resolved.add(extension);

